### ISSUE TITLE
example .pt file, and tools to convert binary image to .pt file?

### ISSUE THREAD

**Issue by dancebean** [getkeops/keops#13]
Hi, I'm interested in testing the KeOps, and am trying to go through the "**Surface registration**" tutorial here:
http://kernel-operations.io/keops/_auto_tutorials/surface_registration/plot_LDDMM_Surface.html#sphx-glr-auto-tutorials-surface-registration-plot-lddmm-surface-py

In the tutorial, it requires the  **"*.pt"** as the import data file:
“**hippos.pt**” : original data (6611 vertices), etc...

I'm assuming it's a point cloud data format. May I ask if there is any example ".pt" file format that we can use to test? Furthermore, would there be tools that can be used to convert a binary mask image to a "**.pt**" file format?

Thank you!

**Comment by bcharlier** [comment]
Hey Ma, thank you for using KeOps!

The data files are in the [./data](https://github.com/getkeops/keops/tree/master/pykeops/tutorials/surface_registration/data) subdir of the script file. Maybe, the best way to start is to clone the entire repo in order to get the example script working 'out-of-th-box'.

**Comment by dancebean** [comment]
Thank you for your help, Benjamin! Yes, sounds good. I'll start by git clone and test the repo.

**Comment by dancebean** [comment]
> 
> 
> Hey Ma, thank you for using KeOps!
> 
> The data files are in the [./data](https://github.com/getkeops/keops/tree/master/pykeops/tutorials/surface_registration/data) subdir of the script file. Maybe, the best way to start is to clone the entire repo in order to get the example script working 'out-of-th-box'.

Hi Benjamin, may I ask for a follow up question that, what tool/function can we use do we create .pt vertices file? And can we convert .pt file to/from numpy, .mat, or FreeSurfer's vertices file? Thank you again!

**Comment by bcharlier** [comment]
No idea... the file was provided by @joanglaunes . Maybe he knows a bit more about this... Let's wait for his answer.

**Comment by joanglaunes** [comment]
Hello @dancebean ,
I cannot find anymore the code I wrote to produce these .pt data file, but what is stores is just the tensors giving the vertex coordinates (V for Vertices) and triangles indices (F for Faces) for both source and target surfaces, thus named VS, FS, VT, FT in the script. 
About FreeSurfer, I do not know it well but there is a tool to load a FreeSurfer mesh here (can be installed via pip) :
https://pypi.org/project/freesurfer-surface/
Then considering you have two FreeSurfer meshes Surf1.pial and Surf2.pial that you want to use as source and target, it should work if you replace the line `VS, FS, VT, FT = torch.load(datafile)` by
```
def load_FreeSurferMesh(datafile):
    import torch
    from freesurfer_surface import Surface, Vertex, Triangle
    surface = Surface.read_triangular(datafile)
    V = torch.tensor(surface.vertices)
    F = torch.zeros(len(surface.triangles),3,dtype=torch.long)
    for i, f in enumerate(surface.triangles):
        F[i,:] = torch.tensor(f.vertex_indices)
    return V, F

VS, FS = load_FreeSurferMesh('Surf1.pial')
VT, FT = load_FreeSurferMesh('Surf2.pial')
```
I only tested this very quickly, I hope it works !

Joan

**Comment by dancebean** [comment]
Hi Benjamin @bcharlier and Joan @joanglaunes . Thank you very much for your kind reply. These are definitely very helpful. The FreeSurfer works great! I'm going to test on self-generated point cloud data as well.
